### PR TITLE
Issue 657: Providing support to update controller service annotations

### LINKED ISSUE #657: Operator is not updating the controller service if new annotations are added
### Description

If we are adding new annotations to controller service, operator is not updating the service

### Importance
 must-have

### Location
pravegacluster_controller.go

### Suggestions for an improvement
Add the functionality to update service if there is changes

### PR DESCRIPTION
**Pull request by anishakj**
### Change log description

Added support to update the annotations and labels at run time for controller service

### Purpose of the change

Fixes #657

### What the code does
In controller reconcile service, if there is change in service like annotations, labels it will be updated

### How to verify it

Verified by adding new annotations at run time

### CODE REVIEW

File: controllers/pravegacluster_controller.go [hunk 1]
```diff
@@ -388,11 +388,29 @@ func (r *PravegaClusterReconciler) reconcileControllerService(p *pravegav1beta1.
...
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(context.TODO(), service)
+			if err != nil && !errors.IsAlreadyExists(err) {
+				return err
+			}
 
+		} else {
+			return err
+		}
+
```
Comment (jkhalack) [inline controllers/pravegacluster_controller.go:403]: (nitpicking) Could you remove the extra empty lines from the new code to conform to the file formatting? Thanks!

Comment (anishakj) [reply controllers/pravegacluster_controller.go]: Done

File: controllers/pravegacluster_controller.go [hunk 2]
```diff
@@ -388,11 +388,28 @@ func (r *PravegaClusterReconciler) reconcileControllerService(p *pravegav1beta1.
...
-		return err
-	}
+	currentService := &corev1.Service{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: service.Name, Namespace: p.Namespace}, currentService)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(context.TODO(), service)
+			if err != nil && !errors.IsAlreadyExists(err) {
+				return err
+			}
```
Comment (jkhalack) [inline controllers/pravegacluster_controller.go:399]: This line still looks strange

Comment (anishakj) [reply controllers/pravegacluster_controller.go:399]: here  are you  suggesting to remove the `IsAlreadyExists` check?

Comment (jkhalack) [reply controllers/pravegacluster_controller.go:399]: I mean the empty line

Comment (anishakj) [reply controllers/pravegacluster_controller.go]: Done

**Review by jkhalack — APPROVED** [review]

**Review by krishan-Rai — APPROVED** [review]

**Pull request merged**